### ISSUE TITLE
0042 - Trapping Rain Water

### ISSUE THREAD

**Issue by vJechsmayr** [vJechsmayr/PythonAlgorithms#181]
## Description of the Problem
Given _n_ non-negative integers representing an elevation map where the width of each bar is 1, compute how much water it is able to trap after raining.

![image](https://user-images.githubusercontent.com/6342890/94930058-283c2c80-04c6-11eb-9951-2e9b02bd0d14.png)
_The above elevation map is represented by array [0,1,0,2,1,0,1,3,2,1,2,1]. In this case, 6 units of rain water (blue section) are being trapped. Thanks Marcos for contributing this image!_

**Example:**
```
Input: [0,1,0,2,1,0,1,3,2,1,2,1]
Output: 6
```

## Code
```
class Solution:
    def trap(self, height: List[int]) -> int:
```

## Link To The LeetCode Problem
[LeetCode](https://leetcode.com/problems/trapping-rain-water/)

**Comment by divyhshah** [comment]
Hey i want to work on this can you please assign this to me

**Comment by divyhshah** [comment]
@vJechsmayr  I am done with this and also added a solution can you please review it
Thanks
closes #181
